vendors/st/boards/stm32l475_discovery/ports/common_io/iot_uart.c

```diff
@@ -0,0 +1,636 @@
+            .HwFlowCtl      = UART_HWCONTROL_NONE,
+            .OverSampling   = UART_OVERSAMPLING_16,
+            .OneBitSampling = UART_ONE_BIT_SAMPLE_DISABLE
+        },
+        .AdvancedInit =
+        {
+            .AdvFeatureInit = UART_ADVFEATURE_NO_INIT
+        },
+        .pTxBuffPtr = NULL,
+        .TxXferSize = 0,
+        .TxXferCount = 0,
+        .pRxBuffPtr = NULL,
+        .RxXferSize = 0,
+        .RxXferCount = 0
+    },
+    {
+        .Instance = UART5,
+        .Init =
+        {
+            .BaudRate       = IOT_UART_BAUD_RATE_DEFAULT,
+            .WordLength     = UART_WORDLENGTH_8B,
+            .StopBits       = UART_STOPBITS_1,
+            .Parity         = UART_PARITY_NONE,
+            .Mode           = UART_MODE_TX_RX,
+            .HwFlowCtl      = UART_HWCONTROL_NONE,
+            .OverSampling   = UART_OVERSAMPLING_16,
+            .OneBitSampling = UART_ONE_BIT_SAMPLE_DISABLE
+        },
+        .AdvancedInit =
+        {
+            .AdvFeatureInit = UART_ADVFEATURE_NO_INIT
+        },
+        .pTxBuffPtr = NULL,
+        .TxXferSize = 0,
+        .TxXferCount = 0,
+        .pRxBuffPtr = NULL,
+        .RxXferSize = 0,
+        .RxXferCount = 0
+    },
+};
+
+static IotUARTDescriptor_t xUart0 =
+{
+    .pvUserCallbackContext = NULL,
+    .pxHuart               = &xUartHandleMap[ 0 ],
+    .eIrqNum               = USART1_IRQn,
+    .xUartCallback         = NULL,
+    .xSemphr               = NULL,
+    .sOpened               = IOT_UART_CLOSED,
+};
+
+static IotUARTDescriptor_t xUart1 =
+{
+    .pvUserCallbackContext = NULL,
+    .pxHuart               = &xUartHandleMap[ 1 ],
+    .eIrqNum               = USART2_IRQn,
+    .xUartCallback         = NULL,
+    .xSemphr               = NULL,
+    .sOpened               = IOT_UART_CLOSED,
+};
+
+static IotUARTDescriptor_t xUart2 =
+{
+    .pvUserCallbackContext = NULL,
+    .pxHuart               = &xUartHandleMap[ 2 ],
+    .eIrqNum               = USART3_IRQn,
+    .xUartCallback         = NULL,
+    .xSemphr               = NULL,
+    .sOpened               = IOT_UART_CLOSED,
+};
+
+static IotUARTDescriptor_t xUart3 =
+{
+    .pvUserCallbackContext = NULL,
+    .pxHuart               = &xUartHandleMap[ 3 ],
+    .eIrqNum               = UART4_IRQn,
+    .xUartCallback         = NULL,
+    .xSemphr               = NULL,
+    .sOpened               = IOT_UART_CLOSED,
+};
+
+static IotUARTDescriptor_t xUart4 =
+{
+    .pvUserCallbackContext = NULL,
+    .pxHuart               = &xUartHandleMap[ 4 ],
+    .eIrqNum               = UART5_IRQn,
+    .xUartCallback         = NULL,
+    .xSemphr               = NULL,
+    .sOpened               = IOT_UART_CLOSED,
+};
+
+static IotUARTHandle_t const pxUarts[] = { &xUart0, &xUart1, &xUart2, &xUart3, &xUart4 };
+
+IotUARTHandle_t iot_uart_open( int32_t lUartInstance )
+{
+    IotUARTHandle_t xHandle = NULL;
+
+    if( ( lUartInstance < IOT_USART_PORTS ) && ( lUartInstance >= 0 ) )
```

Comment:
The other ports seem to be using <= to the total ports, i think we should align all three. This is not mentioned in the header file, so it seems to be up to the implementation as to which one to use

Reply:
What do you mean? instance id should less than total instances number.

Reply:
i2c and spi use <=, so those should then be changed to < if that is the case

Reply:
I see what you are saying now. Fixed them.